peachjam_ml/handler.py

```diff
@@ -1,6 +1,7 @@
 def modify_document_detail_context(sender, context, view, **kwargs):
     if not context["show_related_documents"]:
-        if view.request.user.has_perm("peachjam_ml.view_documentembedding") and hasattr(
-            context["document"], "embedding"
-        ):
+        if (
+            not view.request.user.is_authenticated
+            or view.request.user.has_perm("peachjam_ml.view_documentembedding")
+        ) and hasattr(context["document"], "embedding"):
```

Comment:
is it possible to do this entirely in the template? I don't like the fact that we spreading this logic to this module,